tests/test_data.py

```diff
@@ -344,6 +381,14 @@ def test_get_hdf_dictionary(self):
         self.assertIn('group_data', result['group1'])
         print(result)
 
+    def test_list_root_groups(self):
+        testfile = 'load_hdf_contents_test.hdf'
+        group_data_dict = {'group_data': np.arange(1000)}
+        save_hdf(write_dir, testfile, data_dict=group_data_dict, data_group="/group2", append=True)
+        group_names = list_root_groups(write_dir, testfile)
+        self.assertIn('group1', group_names)
```

Comment:
i don't think there's any guarantee that the test above will run before this test. you should make a completely new file and add both groups in this function to make sure they are there.